Xamarin.Essentials/DeviceInfo/DeviceInfo.shared.cs

```diff
@@ -21,6 +25,53 @@ public static partial class DeviceInfo
         public static DeviceType DeviceType => GetDeviceType();
     }
 
+    public readonly struct StorageInfo : IEquatable<StorageInfo>
+    {
+        public StorageInfo(ulong capapcity, ulong freeBytes, ulong usedBytes, StorageType type)
+        {
+            Capacity = capapcity;
+            FreeBytes = freeBytes;
+            UsedBytes = usedBytes;
+            Type = type;
+        }
+
+        public ulong Capacity { get; }
+
+        public ulong FreeBytes { get; }
+
+        public ulong UsedBytes { get; }
+
+        public StorageType Type { get; }
+
+        public bool Equals(StorageInfo other) =>
+            FreeBytes == other.FreeBytes &&
+            Capacity == other.Capacity &&
+            UsedBytes == other.UsedBytes &&
+            Type == other.Type;
+
+        public static bool operator ==(StorageInfo lhs, StorageInfo rhs) => lhs.Equals(rhs);
+
+        public static bool operator !=(StorageInfo lhs, StorageInfo rhs) => !(lhs == rhs);
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+                return false;
+            return obj is StorageInfo info && Equals(info);
+        }
+
+        public override int GetHashCode() => (Capacity, UsedBytes, FreeBytes, Type).GetHashCode();
+
+        public override string ToString() => $"Capacity: {Capacity.ToString()}{Environment.NewLine}FreeBytes: {FreeBytes.ToString()}{Environment.NewLine}UsedBytes: {UsedBytes.ToString()}{Environment.NewLine}Type: {Type}";
```

Comment:
I would follow the pattern here with no newlines, comma separated sections:

https://github.com/xamarin/Essentials/blob/master/Xamarin.Essentials/Types/DisplayInfo.shared.cs#L46-L49